core/src/main/java/com/yahoo/imapnio/async/data/MessageNumberSet.java

```diff
@@ -257,4 +257,41 @@ public static String buildString(@Nullable final MessageNumberSet[] msgsets) {
         }
         return s.toString();
     }
+
+    /**
+     * Converts an IMAP RFC3501 sequence-set syntax into an array of MessageNumberSet.
+     *
+     * @param msgNumbers the message numbers string
+     * @return the array of MessageNumberSet
+     * @throws ImapAsyncClientException will not throw
+     */
+    public static MessageNumberSet[] buildMessageNumberSets(@Nullable final String msgNumbers) throws ImapAsyncClientException {
+        if (msgNumbers == null || msgNumbers.isEmpty()) {
+            return null;
+        }
+
+        final String[] msgSetStrs = msgNumbers.split(",");
+        final MessageNumberSet[] msgSets = new MessageNumberSet[msgSetStrs.length];
+
+        int i = 0; // msgset index
+        for (final String element : msgSetStrs) {
+            if (element.contains(":")) {
+                final String[] elements = element.split(":");
+                if (elements[1].equals("*")) { // Ex: 1:*
```

Comment:
When do we need this? What happens to *:1 ? Lets go through the ABNF.  Please write unit test that takes the output of buildString() and ensure the output is same as input of buildString().